test/support/pulp/repository_support.rb

```diff
@@ -43,6 +43,13 @@ def self.create_repo(repo_id)
                                path: @repo.relative_path,
                                with_importer: true
                               )
+
+      if @repo.puppet?
+        ForemanTasks.sync_task(::Actions::Pulp::Repository::DistributorPublish,
```

Comment:
working around this bug: https://pulp.plan.io/issues/1607

Its unlikely to be hit in practice because we always publish the repo